aiida/backends/tests/cmdline/commands/test_import.py

```diff
@@ -85,8 +88,67 @@ def test_comment_mode(self):
+        self.assertEqual(result.exit_code, 0, result.output)
 
         options = ['--comment-mode', 'overwrite'] + archives
         result = self.cli_runner.invoke(cmd_import.cmd_import, options)
         self.assertIsNone(result.exception, result.output)
         self.assertIn('Comment mode: overwrite', result.output)
+        self.assertEqual(result.exit_code, 0, result.output)
+
+    def test_import_old_local_archives(self):
+        """ Test import of old local archives
+        Expected behavior: Return message in terminal and error code != 0
+        """
+        archives = [('export/migrate/export_v0.1.aiida', '0.1'), ('export/migrate/export_v0.2.aiida', '0.2'),
+                    ('export/migrate/export_v0.3.aiida', '0.3')]
+
+        for archive, version in archives:
+            options = [get_archive_file(archive)]
+            result = self.cli_runner.invoke(cmd_import.cmd_import, options)
+
+            self.assertIsNotNone(result.exception, result.output)
+            self.assertNotEqual(result.exit_code, 0, result.output)
+            self.assertIn(version, result.output, result.exception)
+
+    def test_import_old_url_archives(self):
+        """ Test import of old URL archives
+        Expected behavior: Return message in terminal and error code != 0
+        """
+        url = "https://raw.githubusercontent.com/aiidateam/aiida_core/f685844b290ed23df254873df62da7162bde0f1b/"
+        archive_path = "aiida/backends/tests/fixtures/export/migrate/"
+        archives = [('export_v0.1.aiida', '0.1'), ('export_v0.2.aiida', '0.2'), ('export_v0.3.aiida', '0.3')]
+
+        for archive, version in archives:
+            options = [url + archive_path + archive]
+            result = self.cli_runner.invoke(cmd_import.cmd_import, options)
+
+            self.assertIsNotNone(result.exception, result.output)
+            self.assertNotEqual(result.exit_code, 0, result.output)
+            self.assertIn(version, result.output, result.exception)
+
+    @unittest.skip("Reenable when issue #2426 has been solved (migrate exported files from 0.3 to 0.4)")
+    def test_import_url_and_local_archive(self):
+        """Test import of both a remote and local archive
+        NOTE: Update 'url' to point at correct commit.
+        """
+        url = "https://raw.githubusercontent.com/yakutovicha/aiida_core/f5fff1846a62051b898f13db67f5eef18892d5f4/"
+        url_archive = "aiida/backends/tests/fixtures/export/migrate/export_v0.4_no_UPF.aiida"
+        local_archive = "export/migrate/export_v0.4_no_UPF.aiida"
+
+        options = [get_archive_file(local_archive), url + url_archive, get_archive_file(local_archive)]
+        result = self.cli_runner.invoke(cmd_import.cmd_import, options)
+
+        self.assertIsNone(result.exception, result.output)
+        self.assertEqual(result.exit_code, 0, result.output)
+
+    def test_import_url_timeout(self):
+        """Test a timeout to valid URL is correctly errored"""
```

Comment:
hm... this I find a but problematic.
the rule of thumb for unit tests is that they should run "in the blink of an eye".
are you waiting for a timeout here? is there a way to reduce this test to ~100ms?

Reply:
Yes, I agree about this. But I want/need to test that we can get to this point in the code. If you have a better suggestion I'm happy to hear it. The timeout is otherwise "only" 10 seconds.

Reply:
I've thought of something. I'll make `timeout` a property exactly for this purpose. Then we can test it with `timeout = 0`.